xinference/device_utils.py

```diff
@@ -0,0 +1,106 @@
+# Copyright 2022-2023 XProbe Inc.
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import os
+
+import torch
+from typing_extensions import Literal, Union
+
+try:
+    import intel_extension_for_pytorch  # noqa: F401
```

Comment:
What's this import do?

Reply:
Intel GPU needs this to enable `torch.xpu` device. See https://intel.github.io/intel-extension-for-pytorch/xpu/latest/tutorials/getting_started.html for more details.

Reply:
As the doc say:
> Important: It is highly recommended to import intel_extension_for_pytorch right after import torch, prior to importing other packages.

This can only be ensured by doing it in the xinference/__init__.py file, does it matter during your experiments?